src/Engine/Managers/SystemDisplay/SystemDisplay.cpp

```diff
@@ -25,11 +32,13 @@ namespace Ra
 #ifndef RA_DISABLE_DEBUG_DISPLAY
         DebugComponent* SystemEntity::dbgCmp()
         {
-            CORE_ASSERT(getInstance()->getNumComponents() == 2,  
-                        "This entity should have two components only");
-            CORE_ASSERT(getInstance()->getTransformAsMatrix() == 
+            SystemEntity *instance=getInstance();
+            /// And here also ;) of course
+            CORE_ASSERT(instance->getNumComponents() == 2 || instance->getNumComponents() == 3,
```

Comment:
Why 2 or 3 ? What changed here ?

Reply:
This is due to the MeshFeatureTrackingPlugin which adds the drawn sphere to the SystemEntity.
Since there was supposed to be only 2 components, I changed this to "2 or 3".
This might not be the best way to do it though.

Reply:
Yeah in fact it's more a reminder that something is "badly designed" than a todo.
Either one (i.e. Plugins) can add some component in SystemEntity so we do not need to check the number of components, either it is not allowed at all (but allow for "my nice plugin only" is not a good idea I think).
But since I'm not the author of theses modifications, I added the todo as a note for later. 
Feel free to change comment, remove the assert, change the way systemEntity work ;) but please at least accept the core_assert test modification so that we can use helper function again (right now if you try  RA_DISPLAY_SPHERE for instance, you have a assert false ! ;) )

Reply:
Sure, ok.
I don't really mind the practical changes in the code, but I would like we keep the code understandable and self contained.
For me, the comments are right now more confusing than helpful.

So please update the comment, such that it explains what is the problem and, if any, clearly state what are the todos.